### PR TITLE
feat(leaky-bucket): adds additional attachment for slack

### PR DESCRIPTION
**Pull request by scefali**
This PR adds a manager to add an additional attachment to Slack issue notifications. Note that this attachment is only added for issue alerts instead of all types of Slack alerts for now. I plan on adding it for messages coming from the notification platform in a future PR.

Related PR: https://github.com/getsentry/getsentry/pull/6717

### CODE REVIEW

File: src/sentry/integrations/slack/notify_action.py [hunk 1]
```diff
@@ -107,6 +108,8 @@ def clean(self) -> Mapping[str, Any]:
             )
 
         channel = cleaned_data.get("channel", "")
+        timed_out = False
+        channel_prefix = ""
```
Comment (scefali) [inline src/sentry/integrations/slack/notify_action.py:112]: Needed these changes to fix a lint error

**Review by NisanthanNanthakumar — APPROVED** [review]

**Comment by mgaeta** [comment]
I like what this is doing but I'd suggest one more thing:
- make `register_additional_attachment_generator()` a decorator
- make `attachment_generator` a list

That way you can do something in getsentry like:
```py
@register_additional_attachment_generator(priority=1.0)
def upsell_attachment():
	...

@register_additional_attachment_generator(priority=0.1)
def quota_warning_attachment():
	...
```

**Comment by scefali** [comment]
> ```python
> quota_warning_attachment
> ```

@mgaeta I like the idea of doing it as a decorator, but I think making `attachment_generator` a list is overkill for what's needed now. We can always convert it to a list later without changing the APIs.

**Comment by mgaeta** [comment]
One more thing I'd do now is move the attachment manager to `src/sentry/notifications/` so that it can start working automatically with other providers.

**Comment by mgaeta** [comment]
If `attachment_generator` is going to be a singleton then you could raise and exception in `register_additional_attachment_generator()` when `attachment_generator` is already set.

**Comment by scefali** [comment]
> One more thing I'd do now is move the attachment manager to `src/sentry/notifications/` so that it can start working automatically with other providers.

@mgaeta Hmm...but this needs to work outside of the notification platform since issue alerts don't use the notification.

**Comment by scefali** [comment]
@mgaeta Ok, I switched to using a decorator with a pattern similar to `register_notification_provider`

**Review by mgaeta — APPROVED** [review]

**Pull request merged**